kotlin-spark-api/src/main/kotlin/org/jetbrains/kotlinx/spark/api/RddDouble.kt

```diff
@@ -0,0 +1,106 @@
+package org.jetbrains.kotlinx.spark.api
+
+import org.apache.spark.api.java.JavaDoubleRDD
+import org.apache.spark.api.java.JavaRDD
+import org.apache.spark.partial.BoundedDouble
+import org.apache.spark.partial.PartialResult
+import org.apache.spark.rdd.RDD
+import org.apache.spark.util.StatCounter
+import scala.Tuple2
+
+@Suppress("UNCHECKED_CAST")
+fun JavaRDD<Double>.toJavaDoubleRDD(): JavaDoubleRDD =
+    JavaDoubleRDD.fromRDD(rdd() as RDD<Any>)
+
+@Suppress("UNCHECKED_CAST")
+fun JavaDoubleRDD.toDoubleRDD(): JavaRDD<Double> =
+    JavaDoubleRDD.toRDD(this).toJavaRDD() as JavaRDD<Double>
+
+/** Add up the elements in this RDD. */
+fun JavaRDD<Double>.sum(): Double = toJavaDoubleRDD().sum()
+
+/**
+ * Return a [org.apache.spark.util.StatCounter] object that captures the mean, variance and
+ * count of the RDD's elements in one operation.
+ */
+fun JavaRDD<Double>.stats(): StatCounter = toJavaDoubleRDD().stats()
+
+/** Compute the mean of this RDD's elements. */
+fun JavaRDD<Double>.mean(): Double = toJavaDoubleRDD().mean()
```

Comment:
Is there mean/average for other types than Double?

Reply:
Nope, although there is the implicit number conversion in scala. In Java there's just `JavaDoubleRDD` and in Scala `DoubleRDDFunctions` for `RDD[Double]`, but it appears I can call the same functions with an `RDD[Int]`... I'll see if I can add conversions for any `Number`!

Reply:
good one actually!